modules/mobile-agents/android/client/client/src/main/java/org/wso2/emm/agent/api/ApplicationManager.java

```diff
@@ -183,10 +185,18 @@ public void uninstallApplication(String packageName, String schedule) {
 		if (Constants.SYSTEM_APP_ENABLED) {
 			CommonUtils.callSystemApp(context, Constants.Operation.SILENT_UNINSTALL_APPLICATION, schedule, packageName);
 		} else {
-			Uri packageURI = Uri.parse(packageName);
-			Intent uninstallIntent = new Intent(Intent.ACTION_DELETE, packageURI);
-			uninstallIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-			context.startActivity(uninstallIntent);
+			if (schedule != null && !schedule.trim().isEmpty() && !schedule.equals("undefined")) {
```

Comment:
In case the system app is enabled, "schedule" is passes thru to the system service and the null/empty/undefined check happens on the system service end. Therefore, instant install and scheduling happen as expected in system app as well.